corehq/motech/repeaters/tests/test_couchsqlmigration.py

```diff
@@ -461,6 +461,16 @@ def test_migrate_record_with_no_attempts(self):
         obj = SQLRepeatRecord.objects.get(couch_id=doc_id)
         self.assertFalse(obj.attempts)
 
+    def test_migrate_record_erroneous_next_check(self):
+        doc, _ = self.create_repeat_record()
+        doc.update(succeeded=True, next_check=datetime.utcnow().isoformat() + 'Z')
+        doc_id = self.db.save_doc(doc)["id"]
+        with templog() as log, patch.object(transaction, "atomic", atomic_check):
+            call_command('populate_repeatrecords', log_path=log.path)
+            self.assertNotIn('has differences:', log.content)
```

Comment:
I'm new to this code. What exactly is the need for this assertion in these tests? It seems like the other assert is what is really testing the intention of this test (that a record is successfully migrated with next_check set to None if we didn't like the format).

Reply:
This assertion simply verifies that the migration did not result in unexpected differences. You are correct that the other assert is the most important.